rest/server_context.go

```diff
@@ -518,8 +525,16 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(ctx context.Context, config
 		useViews = true
 	}
 
+	type indexInitData struct {
+		scopeAndCollection base.ScopeAndCollectionName
+		indexSet           db.CollectionIndexesType
+		datastore          base.DataStore
+	}
+
+	collectionsRequiringIndexes := make([]indexInitData, 0)
+
 	// initDataStore is a function to initialize Views or GSI indexes for a datastore
-	initDataStore := func(ds base.DataStore, metadataIndexes db.CollectionIndexesType) (err error) {
+	initDataStore := func(ds base.DataStore, metadataIndexes db.CollectionIndexesType, name base.ScopeAndCollectionName) (err error) {
```

Comment:
you can derive scope and collection information from a datastore, albeit in an annoying way https://github.com/couchbase/sync_gateway/blob/b20d5152bd4979620ce0098d491c5e6e454af2db/base/bucket.go#L97